bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/ResumingStreamingResultScanner.java

```diff
@@ -87,25 +103,82 @@ public FlatRow next() throws IOException {
       try {
         FlatRow result = currentDelegate.next();
         if (result != null) {
-          retryHandler.update(result);
+          requestManager.updateLastFoundKey(result.getRowKey());
+          // We've had at least one successful RPC, reset the backoff and retry counter
+          currentErrorBackoff = null;
+          timeoutRetryCount = null;
         }
         return result;
       } catch (ScanTimeoutException rte) {
         closeRpcContext();
         closeCurrentDelegate();
-        ReadRowsRequest newRequest = retryHandler.handleScanTimeout(rte);
+        ReadRowsRequest newRequest = handleScanTimeout(rte);
         currentDelegate = scannerFactory.createScanner(newRequest);
         this.rpcContext = rpcMetrics.timeRpc();
       } catch (IOExceptionWithStatus ioe) {
         closeRpcContext();
         closeCurrentDelegate();
-        ReadRowsRequest newRequest = retryHandler.handleIOException(ioe);
+        ReadRowsRequest newRequest = handleIOException(ioe);
         currentDelegate = scannerFactory.createScanner(newRequest);
         this.rpcContext = rpcMetrics.timeRpc();
       }
     }
   }
 
+
+  private ReadRowsRequest handleScanTimeout(ScanTimeoutException rte) throws IOException {
+    logger.info("The client could not get a response in %d ms. Retrying the scan.",
+      retryOptions.getReadPartialRowTimeoutMillis());
+
+    // Reset the error backoff in case we encountered this timeout after an error.
+    // Otherwise, we will likely have already exceeded the max elapsed time for the backoff
+    // and won't retry after the next error.
+    currentErrorBackoff = null;
```

Comment:
similar to timeoutRetryCount, might be less error prone to construct once & reset when necessary